atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/TimeLockServerTest.java

```diff
@@ -15,102 +15,52 @@
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Ignore;
-import org.junit.Test;
-
 import com.google.common.collect.ImmutableSet;
-import com.palantir.atlasdb.timelock.config.ImmutableAtomixConfiguration;
 import com.palantir.atlasdb.timelock.config.ImmutableClusterConfiguration;
+import com.palantir.atlasdb.timelock.config.TimeLockAlgorithmConfiguration;
 import com.palantir.atlasdb.timelock.config.TimeLockServerConfiguration;
 
-import io.atomix.catalyst.transport.Address;
-import io.atomix.copycat.server.storage.StorageLevel;
-import io.dropwizard.jersey.setup.JerseyEnvironment;
-import io.dropwizard.lifecycle.setup.LifecycleEnvironment;
-import io.dropwizard.setup.Environment;
-
-@Ignore("Observed ConcurrentModificationException-related flakes (e.g. build #5407 on CircleCI)."
-        + "Fixed in atomix/copycat#231, but not part of Copycat 1.1.4 which we use.")
-public class TimeLockServerTest {
-    private static final String LOCAL_ADDRESS_STRING = "localhost:12345";
-    private static final Address LOCAL_ADDRESS = new Address(LOCAL_ADDRESS_STRING);
-    private static final TimeLockServerConfiguration TIMELOCK_CONFIG = new TimeLockServerConfiguration(
-            ImmutableAtomixConfiguration.builder()
-                    .storageLevel(StorageLevel.MEMORY)
-                    .build(),
-            ImmutableClusterConfiguration.builder()
-                    .localServer(LOCAL_ADDRESS_STRING)
-                    .addServers(LOCAL_ADDRESS_STRING)
-                    .build(),
-            ImmutableSet.of(String.format("%s:%s", LOCAL_ADDRESS.host(), LOCAL_ADDRESS.port())));
-
-    private final TimeLockServer server = new TimeLockServer();
-    private final Environment environment = mock(Environment.class);
-    private final List<LifeCycle.Listener> listeners = new ArrayList<>();
-
-    @Before
-    public void setupEnvironment() {
-        when(environment.jersey()).thenReturn(mock(JerseyEnvironment.class));
-
-        LifecycleEnvironment lifecycle = mock(LifecycleEnvironment.class);
-        when(environment.lifecycle()).thenReturn(lifecycle);
-        doAnswer(inv -> listeners.add((LifeCycle.Listener) inv.getArguments()[0]))
-                .when(lifecycle).addLifeCycleListener(any());
-    }
-
-    @After
-    public void sendShutdownToListeners() {
-        LifeCycle event = mock(LifeCycle.class);
-        listeners.forEach(listener -> listener.lifeCycleStopping(event));
-        listeners.forEach(listener -> listener.lifeCycleStopped(event));
+public class TimeLockServerTest extends ServerImplementationTest {
+    private static final String LOCAL_ADDRESS = "localhost:8080";
+    private static final String TEST_CLIENT = "test";
+
+    private ServerImplementation serverImplementation;
+
+    @Override
+    protected TimeLockServerConfiguration getConfiguration() {
+        TimeLockAlgorithmConfiguration algorithmConfiguration = mock(TimeLockAlgorithmConfiguration.class);
```

Comment:
the ``setup`` should be extracted out and run as a `@Before` method. `getConfiguration ` can be renamed to  `getTimeLockServerConfiguration` and should just return the config.